### PR TITLE
Warn against redefining global variables  in the code editor

### LINKED ISSUE #7270: Profile lookup exercise. Unchecked points although the output is correct
Challenge [Profile Lookup](http://www.freecodecamp.com/challenges/profile-lookup#?solution=%2F%2FSetup%0Avar%20contacts%20%3D%20%5B%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%22firstName%22%3A%20%22Akira%22%2C%0A%20%20%20%20%20%20%20%20%22lastName%22%3A%20%22Laine%22%2C%0A%20%20%20%20%20%20%20%20%22number%22%3A%20%220543236543%22%2C%0A%20%20%20%20%20%20%20%20%22likes%22%3A%20%5B%22Pizza%22%2C%20%22Coding%22%2C%20%22Brownie%20Points%22%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%22firstName%22%3A%20%22Harry%22%2C%0A%20%20%20%20%20%20%20%20%22lastName%22%3A%20%22Potter%22%2C%0A%20%20%20%20%20%20%20%20%22number%22%3A%20%220994372684%22%2C%0A%20%20%20%20%20%20%20%20%22likes%22%3A%20%5B%22Hogwarts%22%2C%20%22Magic%22%2C%20%22Hagrid%22%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%22firstName%22%3A%20%22Sherlock%22%2C%0A%20%20%20%20%20%20%20%20%22lastName%22%3A%20%22Holmes%22%2C%0A%20%20%20%20%20%20%20%20%22number%22%3A%20%220487345643%22%2C%0A%20%20%20%20%20%20%20%20%22likes%22%3A%20%5B%22Intriguing%20Cases%22%2C%20%22Violin%22%5D%0A%20%20%20%20%7D%2C%0A%20%20%20%20%7B%0A%20%20%20%20%20%20%20%20%22firstName%22%3A%20%22Kristian%22%2C%0A%20%20%20%20%20%20%20%20%22lastName%22%3A%20%22Vos%22%2C%0A%20%20%20%20%20%20%20%20%22number%22%3A%20%22unknown%22%2C%0A%20%20%20%20%20%20%20%20%22likes%22%3A%20%5B%22Javascript%22%2C%20%22Gaming%22%2C%20%22Foxes%22%5D%0A%20%20%20%20%7D%0A%5D%3B%0A%0Avar%20name%20%3D%20%5B%22Akira%22%2C%22Harry%22%2C%22Sherlock%22%2C%22Kristian%22%5D%3B%0Avar%20propDefault%3D%5B%22firstName%22%2C%22lastName%22%2C%22number%22%2C%22likes%22%5D%3B%0Afunction%20lookUp%28firstName%2C%20prop%29%7B%0A%20%20var%20a%3Dname.indexOf%28firstName%29%3B%0A%20%20var%20b%3DpropDefault.indexOf%28prop%29%3B%0A%20%20if%28a%3E-1%26%26b%3E-1%29%7B%0A%20%20%20%20return%20contacts%5Ba%5D%5Bprop%5D%3B%0A%20%20%7D%0A%20%20if%28a%3D%3D-1%29%7B%0A%20%20%20%20return%20%22No%20such%20contact%22%3B%0A%20%20%7D%0A%20%20if%28b%3D%3D-1%29%7B%0A%20%20%20%20return%20%22No%20such%20property%22%3B%0A%20%20%7D%0A%20%20%0A%7D%0A%0AlookUp%28%22Sherlock%22%2C%20%22likes%22%20%29%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.116 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:

``` javascript
//Setup
var contacts = [
    {
        "firstName": "Akira",
        "lastName": "Laine",
        "number": "0543236543",
        "likes": ["Pizza", "Coding", "Brownie Points"]
    },
    {
        "firstName": "Harry",
        "lastName": "Potter",
        "number": "0994372684",
        "likes": ["Hogwarts", "Magic", "Hagrid"]
    },
    {
        "firstName": "Sherlock",
        "lastName": "Holmes",
        "number": "0487345643",
        "likes": ["Intriguing Cases", "Violin"]
    },
    {
        "firstName": "Kristian",
        "lastName": "Vos",
        "number": "unknown",
        "likes": ["Javascript", "Gaming", "Foxes"]
    }
];

var name = ["Akira","Harry","Sherlock","Kristian"];
var propDefault=["firstName","lastName","number","likes"];
function lookUp(firstName, prop){
  var a=name.indexOf(firstName);
  var b=propDefault.indexOf(prop);
  if(a>-1&&b>-1){
    return contacts[a][prop];
  }
  if(a==-1){
    return "No such contact";
  }
  if(b==-1){
    return "No such property";
  }

}

lookUp("Sherlock", "likes" );

```

### PR DESCRIPTION
**Pull request by Bouncey**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #7270

#### Description

Adds `window` prop names to a `predef` list used for codemirror linting.

Linter warns that you are 'redefining ${variableName}'

### CODE REVIEW

File: common/app/routes/challenges/views/classic/Editor.jsx [hunk 1]
```diff
@@ -11,9 +11,12 @@ import ns from './ns.json';
 import CodeMirrorSkeleton from '../../Code-Mirror-Skeleton.jsx';
 
 const editorDebounceTimeout = 750;
-
+const envProps = typeof window !== 'undefined' ? Object.keys(window) : [];
 const options = {
-  lint: {esversion: 6},
+  lint: {
+    esversion: 6,
+    predef: [ ...envProps ]
```
Comment (BerkeleyTrue) [inline common/app/routes/challenges/views/classic/Editor.jsx]: You can just you the `envProps` direction.

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

**Review by BerkeleyTrue — APPROVED** [review]

**Comment by BerkeleyTrue** [comment]
Code LGTM.

**Comment by Bouncey** [comment]
cc @freeCodeCamp/moderators

**Review by Greenheart — APPROVED** [review]

**Comment by systimotic** [comment]
@Bouncey Can you update this to resolve the merge conflict?

**Comment by BerkeleyTrue** [comment]
merged!!!!!
https://github.com/freeCodeCamp/freeCodeCamp/commit/1126bf7004ec6cabfeb0641eefc823175d33c4ba

🍺 🍻 

Happy Coding!

**Pull request closed without merging**